src/west/configuration.py

```diff
@@ -173,24 +179,26 @@ def __init__(self, topdir: PathType | None = None):
         :param topdir: workspace location; may be None
         '''
 
-        local_path = _location(ConfigFile.LOCAL, topdir=topdir, find_local=False) or None
+        local_paths = _location(ConfigFile.LOCAL, topdir=topdir, find_local=False) or None
 
-        self._system_path = Path(_location(ConfigFile.SYSTEM))
-        self._global_path = Path(_location(ConfigFile.GLOBAL))
-        self._local_path = Path(local_path) if local_path is not None else None
+        self._system_paths = str_to_paths(_location(ConfigFile.SYSTEM))
+        self._global_paths = str_to_paths(_location(ConfigFile.GLOBAL))
+        self._local_paths = str_to_paths(local_paths)
 
-        self._system = _InternalCF.from_path(self._system_path)
-        self._global = _InternalCF.from_path(self._global_path)
-        self._local = _InternalCF.from_path(self._local_path)
+        self._system = _InternalCF.from_paths(self._system_paths)
+        self._global = _InternalCF.from_paths(self._global_paths)
+        self._local = _InternalCF.from_paths(self._local_paths)
 
-    def get_paths(self, location: ConfigFile = ConfigFile.ALL):
+    def get_paths(self, location: ConfigFile = ConfigFile.ALL, only_existing=True) -> list[Path]:
```

Comment:
I think `only_existing=False` should be the default to not change the function behavior. Also `existing_only` reads a bit nicer, IMO?